sunpy/net/dataretriever/sources/kanzelhohe.py

```diff
@@ -0,0 +1,132 @@
+    Instrument: Fixed argument = 'kanzelhohe'
+    Wavelength: Fixed argument = astropy.units.quantity.Quantity
+                The physical value of wavelength will belong to any of [5460, 6563, 32768]
+                and units will be Angstroms.
+    Returns
+    -------
+    urls: list
+    list of urls corresponding to requested time range.
+    Examples
+    --------
+    >>> from sunpy.net import Fido, attrs as a
+    >>> timerange = a.Time('2015/12/28 00:00:00','2015/12/28 09:03:00')
+    >>> results = Fido.search(timerange, a.Instrument('kanzelhohe'),
+    ...                       a.Wavelength(6563*u.AA))   #doctest: +REMOTE_DATA
+    >>> print(results)  #doctest: +REMOTE_DATA
+    Results from 1 Provider:
+    <BLANKLINE>
+    1 Results from the KanzelhoheClient:
+         Start Time           End Time      ...   Instrument      Wavelength
+           str19               str19        ...     str14           str15
+    ------------------- ------------------- ... -------------- ---------------
+    2015-12-28 00:00:00 2015-12-28 09:03:00 ... Kanzelhohe HA2 6563.0 Angstrom
+    <BLANKLINE>
+    <BLANKLINE>
+    >>> response = Fido.fetch(results)  #doctest: +SKIP
+    """
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Return list of URLS corresponding to value of input timerange.
+        Parameters
+        ----------
+        timerange: `sunpy.time.TimeRange`
+            time range for which data is to be downloaded.
+        Returns
+        -------
+        urls : list
+            list of URLs corresponding to the requested time range
+        """
+        wave = kwargs['wavelength']
+        wave_float = (wave.min if isinstance(wave, a.Wavelength) else wave.min()).value
+        table = {6563: ['halpha2k/recent', 'halph_fr'],
+                 32768: ['caiia', 'caiik_fi'],
+                 5460: ['phokada', 'bband_fi']}
+        # Checking if value is close enough to a wavelength value.
+        # Converting from one unit to other introduces precision errors.
+        try:
+            # The isclose function in numpy can only check two arrays if they
+            # are close to each other within some precision. Standalone values such
+            # as int, doubles etc can't be checked that way. In order to do that, we put the
+            # two indiviual values in two seperate arrays da and db and then apply
+            # isclose on both those arrays.
+            da = list()
+            da.append(wave_float)
+            for wave_nums in table.keys():
+                db = list()
+                db.append(wave_nums)
+                if np.isclose(da, db, 1e-10, 1e-10):
+                    wave = wave_nums
+
+            date_table = {6563: datetime.datetime(2000, 7, 20, 7, 45, 46),
+                          5460: datetime.datetime(2011, 1, 7, 10, 7, 33),
+                          32768: datetime.datetime(2010, 7, 31, 8, 10, 59)}
```

Comment:
I would prefer to have this table merged with `table`, so there's one place where we need to update if they decide to add a new filter to their observations.

Reply:
I didn't understood this one. Is it we should predefine it in `sunpy.database` or we should use `~sunpy.database.table` instead of python dictionary?

Reply:
I mean, that this information about the dates could be in the same dictionary where the names are kept above. So if in the future we need to add a new instrument, we can only change one dictionary and not two.